File: bindings/go/binding_test.go
package tree_sitter_php_test

import (
	"testing"

	tree_sitter "github.com/tree-sitter/go-tree-sitter"
	tree_sitter_php "github.com/tree-sitter/tree-sitter-php/bindings/go"
)

func TestPHPGrammar(t *testing.T) {
	language := tree_sitter.NewLanguage(tree_sitter_php.LanguagePHP())
	if language == nil {
		t.Errorf("Error loading PHP grammar")
	}

	sourceCode := []byte("<?php echo 'Hello, World!';")
	parser := tree_sitter.NewParser()
	defer parser.Close()
	parser.SetLanguage(language)

	tree := parser.Parse(sourceCode, nil)
	if tree == nil || tree.RootNode().HasError() {
		t.Errorf("Error parsing PHP")
	}
}

func TestPHPOnlyGrammar(t *testing.T) {
	language := tree_sitter.NewLanguage(tree_sitter_php.LanguagePHPOnly())
	if language == nil {
		t.Errorf("Error loading PHP-Only grammar")
	}

	sourceCode := []byte("echo 'Hello, World!';")
	parser := tree_sitter.NewParser()
	defer parser.Close()
	parser.SetLanguage(language)

	tree := parser.Parse(sourceCode, nil)
	if tree == nil || tree.RootNode().HasError() {
		t.Errorf("Error parsing PHP")
	}
}

File: bindings/node/binding_test.js
/// <reference types="node" />

const assert = require("node:assert");
const { describe, it } = require("node:test");

const Parser = require("tree-sitter");
const { php, php_only } = require("../..");

describe("PHP", () => {
  const parser = new Parser();
  parser.setLanguage(php);

  it("should be named php", () => {
    assert.strictEqual(parser.getLanguage().name, "php");
  });

  it("should parse source code", () => {
    const sourceCode = "<?php echo 'Hello, World!';";
    const tree = parser.parse(sourceCode);
    assert(!tree.rootNode.hasError);
  });
});

describe("PHP Only", () => {
  const parser = new Parser();
  parser.setLanguage(php_only);

  it("should be named php_only", () => {
    assert.strictEqual(parser.getLanguage().name, "php_only");
  });

  it("should parse source code", () => {
    const sourceCode = "echo 'Hello, World!';";
    const tree = parser.parse(sourceCode);
    assert(!tree.rootNode.hasError);
  });
});
